### DISCUSSION TITLE
how to get hyperdiv.state in the scope of the route functions?

### DISCUSSION THREAD

**Discussion by psgg1981** [hyperdiv/hyperdiv#63] [Help] [answered]
Hi again,

While adapting the app template demo showcased in `hyperdiv docs` I've tried to reuse [state](http://localhost:8988/reference/components/state) within the [route functions](http://localhost:8988/reference/components/router), however I get a `NameError: name 'state' is not defined`. This is due to state being defined only on the main function, I guess. But if I try to define it globally I get a `RuntimeError: There's no active Hyperdiv frame.` Is there a proper way to use the state globally?

...
```
# Define the pages of the app:
@router.route("/myapp/home")
def home():
    hd.h2(f"Welcome, {state.logged_in_user['name']}!")
```
...
```
def main():    
    state = hd.state(logged_in_user=None)
```

Thanks!

**Comment by mariusnita** [comment, marked as answer]
Hi,

To share state across routes you can try using [global state](https://docs.hyperdiv.io/reference/components/global_state).

```py
@hd.global_state
class MyState(hd.BaseState):
  logged_in_user = hd.Prop(hd.Any, None)
  
@router.route("/foo")
def foo():
  s = MyState()
  # do stuff with s.logged_in_user
  
@router.route("/bar")
def bar():
  s = MyState()
  # do stuff with s.logged_in_user
  ```
